site/docs/relations/logical_relations.md

```diff
@@ -235,6 +235,40 @@ The join operation will combine two separate inputs into a single output, based
     ```
 
 
+## AsOfJoin Operation
+
+The as-of join operation is a time series operation that will combine a left input and a right input into a single output, based on a join expression, an `on` field and a tolerance value. All inputs must have the on-field in ascending-order. The operation is similar to a join operation where the join expression is used for exact matching whereas the `on` field is used for inexact matching up to the tolerance value.
+
+| Signature            | Value                                                        |
+| -------------------- | ------------------------------------------------------------ |
+| Inputs               | 1                                                            |
+| Outputs              | 1                                                            |
+| Property Maintenance | Distribution is maintained. Orderedness is by the on-field only post operation. Physical relations may provide better property maintenance. |
+| Direct Output Order  | The emit order of the left input followed by the emit order of the right input. |
```

Comment:
Do we allow de duplication of the on field (and by field)?

Reply:
I think that's something to be done by ibis or something producing ibis expressions, so I don't think that'll be anywhere in the Substrait spec.